store.rb

```diff
@@ -118,4 +153,26 @@ def get_node(node)
       raise e
     end
   end
+
+  # immediately update cache if node joins/leaves
+  def watch()
+    return if @zk.nil?
+    @root_watcher ||= @zk.register("/", :only => :child) do |event|
```

Comment:
Hmm, that's good to know. Much cleaner than the synapse's (which I referenced to) unsubscribe/register approach: https://github.com/airbnb/synapse/blob/master/lib/synapse/service_watcher/zookeeper.rb#L89-L90